src/server.rs

```diff
@@ -1489,6 +1562,9 @@ impl ServerRunner {
             return Ok(());
         }
 
+        // Only when the server has finished shutting down, we are allowed to drop sentinel values.
```

Comment:
Even more misleading, because the previous `return Ok(());` omits this explicit `drop()`.